Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -42,80 +42,38 @@
 #:    the preexisting formula already uses.
 
 require "formula"
+require "cli_parser"
 
 module Homebrew
   module_function
 
-  def inreplace_pairs(path, replacement_pairs)
-    if ARGV.dry_run?
-      contents = path.open("r") { |f| Formulary.ensure_utf8_encoding(f).read }
-      contents.extend(StringInreplaceExtension)
-      replacement_pairs.each do |old, new|
-        unless ARGV.flag?("--quiet")
-          ohai "replace #{old.inspect} with #{new.inspect}"
-        end
-        contents.gsub!(old, new)
-      end
-      unless contents.errors.empty?
-        raise Utils::InreplaceError, path => contents.errors
-      end
-      path.atomic_write(contents) if ARGV.include?("--write")
-      contents
-    else
-      Utils::Inreplace.inreplace(path) do |s|
-        replacement_pairs.each do |old, new|
-          unless ARGV.flag?("--quiet")
-            ohai "replace #{old.inspect} with #{new.inspect}"
-          end
-          s.gsub!(old, new)
-        end
-      end
-      path.open("r") { |f| Formulary.ensure_utf8_encoding(f).read }
-    end
-  end
-
-  def formula_version(formula, spec, contents = nil)
-    name = formula.name
-    path = formula.path
-    if contents
-      Formulary.from_contents(name, path, contents, spec).version
-    else
-      Formulary::FormulaLoader.new(name, path).get_formula(spec).version
-    end
-  end
-
-  def fetch_pull_requests(formula)
-    GitHub.issues_for_formula(formula.name, tap: formula.tap).select do |pr|
-      pr["html_url"].include?("/pull/") &&
-        /(^|\s)#{Regexp.quote(formula.name)}(:|\s|$)/i =~ pr["title"]
-    end
-  rescue GitHub::RateLimitExceededError => e
-    opoo e.message
-    []
-  end
-
-  def check_for_duplicate_pull_requests(formula)
-    pull_requests = fetch_pull_requests(formula)
-    return unless pull_requests
-    return if pull_requests.empty?
-    duplicates_message = <<~EOS
-      These open pull requests may be duplicates:
-      #{pull_requests.map { |pr| "#{pr["title"]} #{pr["html_url"]}" }.join("\n")}
-    EOS
-    error_message = "Duplicate PRs should not be opened. Use --force to override this error."
-    if ARGV.force? && !ARGV.flag?("--quiet")
-      opoo duplicates_message
-    elsif !ARGV.force? && ARGV.flag?("--quiet")
-      odie error_message
-    elsif !ARGV.force?
-      odie <<~EOS
-        #{duplicates_message.chomp}
-        #{error_message}
-      EOS
+  def bump_formula_pr
+    @args = Homebrew::CLI::Parser.parse do
+      switch "--devel"
+      switch "-n", "--dry-run"
+      switch "--write"
+      switch "--audit"
+      switch "--strict"
+      switch "--no-browse"
+      switch :quiet
+      switch :force
+      switch :verbose
+      switch :debug
+
+      flag   "--url",      required: true,
```

Comment:
I find it confusing to say it's `required: true` when it's not required if you've passed --tag.

Reply:
I keep forgetting `required` refers to whether it needs a value not whether the flag is ... required. Maybe it could be `value_required: true`, or even
```
flag "--url="
```
and let it speak for itself.

Reply:
`flag "--url="` seems nice.